### PR TITLE
feat: checkbox to count managed events for team-wide limits

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds a checkbox to team-wide bookings limits so admins can choose if bookings from managed event types are counted for the time-wide booking limits. 

Screenshot 2024-10-03 at 11 48 58 AM

Also fixes the following bug:
Screenshot 2024-10-04 at 8 54 32 PM
How to reproduce bug? 
- Set schedule timezone to Singapore  (any + timezone)
- Add event type limit, 1 per year 
- book one meeting
- load availability again and see that it's still shows slots for this year, when trying to confirm the booking you get an error

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue: https://github.com/calcom/docs/issues/139
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking/global-booking-limits.test.ts [hunk 1]
```diff
@@ -369,10 +429,10 @@ describe(
 
         const { req: req2 } = createMockNextJsRequest({
           method: "POST",
-          body: mockBookingData1,
+          body: mockBookingData2,
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleNewBooking/global-booking-limits.test.ts:432]: the wrong mockBookingData was used here

File: packages/features/bookings/lib/handleNewBooking/global-booking-limits.test.ts [hunk 2]
```diff
@@ -468,10 +557,9 @@ describe(
...
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleNewBooking/global-booking-limits.test.ts:560]: same, wrong mockBookingData was used

File: packages/core/bookingLimits/getBusyTimesFromLimits.ts [hunk 3]
```diff
@@ -123,6 +129,8 @@ const _getBusyTimesFromBookingLimits = async (params: {
             teamId,
             user,
             rescheduleUid,
+            includeManagedEvents,
+            timeZone,
```
Comment (CarinaWolli) [inline packages/core/bookingLimits/getBusyTimesFromLimits.ts:133]: not passing timeZone here caused this error with yearly booking limits: 
Screenshot 2024-10-04 at 9 06 34 PM

**Review by zomars — DISMISSED** [review]
Code LGTM. Waiting on Checks. 🙏🏽

**Review by hbjORbj — APPROVED** [review]

**Review by hbjORbj — APPROVED** [review]

**Pull request merged**